test/realm_test.dart

```diff
@@ -1394,6 +1396,173 @@ Future<void> main([List<String>? args]) async {
+    final beforeCompactSizeSize = await createRealmForCompact(config);
+
+    final receivePort = ReceivePort();
+    await Isolate.spawn((List<Object> args) async {
+      SendPort sendPort = args[0] as SendPort;
+      final path = args[1] as String;
+      var config = Configuration.local([Product.schema], path: path);
+      final compacted = Realm.compact(config);
+      Isolate.exit(sendPort, compacted);
+    }, [receivePort.sendPort, config.path]);
+
+    final compacted = await receivePort.first as bool;
+
+    validateCompact(compacted, config.path, beforeCompactSizeSize);
+
+    //test the realm can be opened.
+    final realm = getRealm(config);
+  });
+
+  test('Realm - local encrypted realm can be compacted', () async {
+    final config = Configuration.local([Product.schema],
+        encryptionKey: generateEncryptionKey(), path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+
+    final beforeCompactSizeSize = await createRealmForCompact(config);
+
+    final compacted = Realm.compact(config);
+
+    validateCompact(compacted, config.path, beforeCompactSizeSize);
+
+    //test the realm can be opened.
+    final realm = getRealm(config);
+  });
+
+  test('Realm - in-memory realm can not be compacted', () async {
+    var config = Configuration.inMemory([Product.schema], path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+    expect(() => Realm.compact(config), throws<RealmException>("Can't compact an in-memory Realm"));
+  });
+
+  test('Realm - readonly realm can not be compacted', () async {
+    var path = p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm");
+    var config = Configuration.local([Product.schema], path: path);
+    final beforeCompactSize = await createRealmForCompact(config);
+
+    config = Configuration.local([Product.schema], isReadOnly: true, path: path);
+    expect(() => Realm.compact(config), throws<RealmException>("Can't compact a read-only Realm"));
+
+    //test the realm can be opened.
+    final realm = getRealm(config);
+  });
+
+  test('Realm - disconnected sync realm can be compacted', () async {
+    var config = Configuration.disconnectedSync([Product.schema], path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+
+    final beforeCompactSize = await createRealmForCompact(config);
+
+    final compacted = Realm.compact(config);
+
+    validateCompact(compacted, config.path, beforeCompactSize);
+
+    //test the realm can be opened.
+    final realm = getRealm(config);
+  });
+
+  baasTest('Realm - synced realm can be compacted', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    final path = p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm");
```

Comment:
We don't have to generate path here. The path is already unique by the user id. FlexibleSyncConfiguration will generate it unique.

Reply:
I don't agree. Is there an issue with generating it.

Reply:
I will leave it as it is.